app/src/main/kotlin/compiler/analysis/TypeChecker.kt

```diff
@@ -173,6 +173,8 @@ class TypeChecker(private val nameResolution: Map<Ref<AstNode>, Ref<NamedNode>>,
                         is Function -> report(TypeCheckingError.FunctionAsValue(expression, node))
                     }
                 }
+                is Expression.ArrayElement -> TODO()
+                is Expression.ArrayLength -> TODO()
```

Comment:
If we add TODOs here, then we should also add them in other analysis classes (e.g. NameResolver, ArgumentResolver...).